### PR TITLE
Add template method for dynamically adding merkle tree config

### PR DESCRIPTION
**Pull request by mmedenjak**
Fixes: https://github.com/hazelcast/hazelcast/issues/13656

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/DynamicConfigTemplate.java [hunk 1]
```diff
@@ -516,4 +516,21 @@ void addFlakeIdGeneratorConfig(String name, int prefetchCount, long prefetchVali
...
+     *                are also matched by pattern and a merkle with map name "default"
+     *                applies to all maps that do not have more specific merkle tree configs.
+     * @param enabled {@code true} to enable this merkle tree configuration, otherwise {@code false}
+     * @param depth   depth of the merkle tree. The depth must be between
+     *                {@value com.hazelcast.config.MerkleTreeConfig#MIN_DEPTH}
+     *                and {@value com.hazelcast.config.MerkleTreeConfig#MAX_DEPTH} (exclusive).
+     */
+    @Request(id = 23, retryable = false, response = ResponseMessageConst.VOID)
+    @Since("1.7")
+    void addMerkleTreeConfig(@Nullable String mapName, boolean enabled, int depth);
```
Comment (sancar) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/DynamicConfigTemplate.java]: What does a null map name means ? Can you document that ?

Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/DynamicConfigTemplate.java]: Copy-paste, fixed. Thanks!

**Comment by mmedenjak** [comment]
@sancar can we merge this and release the client protocol?

**Comment by mmedenjak** [comment]
Thanks for the reviews @sancar @asimarslan !

**Pull request merged**